### DISCUSSION TITLE
Upgrade openDCIM from 4.3 to 18 then to 23.3

### DISCUSSION THREAD

**Discussion by ll15039** [opendcim/openDCIM#1445] [Q&A] [answered]
Thank-you all for the excellent openDCIM product.

Completed upgrade from 4.3 to 18 then 18 to 23.3.  Received no errors in the browser window during conversion.
Appears everything works except viewing multiple cabinets at once.  Can view individual device data and people/contacts.
Selecting single cabinets will show all devices in the cabinet correctly
Selecting multiple cabinet views will show a bouncing square forever.
Did a browser Q-inspect which showed the following error in the console section:

> GET https://opendcim01.admin.company.net/api/v1/cabrow/20/devices  [HTTP/1.1 404 Not Found 23ms]

Again, if selecting ONE cabinet, all devices display correctly, the issue is strictly with displaying multiple cabinets at once.

New server:
RHEL 9
Mysql: 8.0.32
PHP: 8.0.27
Apache: 2.4.53

Thank-you for any help.

**Comment by ArchaicHammer** [comment]
Good morning,

I have the same problem but I was able to resolve the 404 problem by modifying api/v1/getRoutes.php:

```
//
//	URL:	/api/v1/cabrow/:cabrowid/devices
//	Method:	GET
//	Params:	none
//	Returns:  All devices in the cabinet row 
//

$app->get( '/cabrow/{cabrowid}/devices', function(Request $request, Response $response, array $args) {
	$r['error']=false;
	$r['errorcode']=200;
	$r['device']=Device::SearchDevicebyCabRow($args[$cabrowid]);
	return $response->withJson($r, $r['errorcode']);
});
```

But now the javascript does not render correctly and loops on the same Devices. If anyone has a solution, have a nice day :)

**Comment by samilliken** [comment, marked as answer]
Yeah, this one was fixed in https://github.com/opendcim/openDCIM/issues/1435 but hasn't made it into a packaged release, yet.   If you simply replace your api/v1/getRoutes.php file with the version currently on GitHub you will be fine.

**Comment by ll15039** [comment]
Replacing  api/v1/getRoutes.php with the latest version did fix that issue.
